test/spec/modules/timBidAdapter_spec.js

```diff
@@ -62,5 +62,91 @@ describe('timAdapterTests', function () {
       expect(requests[0].url).to.exists;
     });
+
+    it('bidRequest data', function () {
+      const requests = spec.buildRequests(validBidRequests);
+      expect(requests[0].data).to.exists;
+    });
+
+    it('bidRequest options', function () {
+      const requests = spec.buildRequests(validBidRequests);
+      expect(requests[0].options).to.exists;
+    });
+  });
+
+  describe('interpretResponse', function () {
+    const bidRequest = {
+      'method': 'GET',
+      'url': '//hb.stinger-bidder.tech/api/v2/services/prebid/testpublisherid/header-bid-tag-0?br=%7B%22id%22%3A%223a3ac0d7fc2548%22%2C%22imp%22%3A%5B%7B%22id%22%3A%22251b8a6d3aac3e%22%2C%22banner%22%3A%7B%22w%22%3A300%2C%22h%22%3A250%7D%2C%22tagid%22%3A%22header-bid-tag-0%22%7D%5D%2C%22site%22%3A%7B%22domain%22%3A%22www.chinatimes.com%22%2C%22page%22%3A%22http%3A%2F%2Fwww.chinatimes.com%2Fa%22%2C%22publisher%22%3A%7B%22id%22%3A%22testpublisherid%22%7D%7D%2C%22device%22%3A%7B%22language%22%3A%22en%22%2C%22w%22%3A300%2C%22h%22%3A250%2C%22js%22%3A1%2C%22ua%22%3A%22Mozilla%2F5.0%20(Windows%20NT%2010.0%3B%20Win64%3B%20x64)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F71.0.3578.98%20Safari%2F537.36%22%7D%2C%22bidId%22%3A%22251b8a6d3aac3e%22%7D',
+      'data': '',
+      'options': {'withCredentials': false}
+    };
+
+    const serverResponse = {
+      'body': {
+        'id': '3a3ac0d7fc2548',
+        'seatbid': []
+      },
+      'headers': {}
+    };
+
+    it('check empty array response', function () {
+      const result = spec.interpretResponse(serverResponse, bidRequest);
+      expect(result).to.deep.equal([]);
+    });
+
+    const validBidRequest = {
+      'method': 'GET',
+      'url': '//hb.stinger-bidder.tech/api/v2/services/prebid/testpublisherid/placementCodeTest?br=%7B%22id%22%3A%2248640869bd9db94%22%2C%22imp%22%3A%5B%7B%22id%22%3A%224746fcaa11197f3%22%2C%22banner%22%3A%7B%22w%22%3A300%2C%22h%22%3A250%7D%2C%22tagid%22%3A%22placementCodeTest%22%7D%5D%2C%22site%22%3A%7B%22domain%22%3A%22mediamart.tv%22%2C%22page%22%3A%22http%3A%2F%2Fmediamart.tv%2Fsas%2Ftests%2FDesktop%2Fcaesar%2Fdfptest.html%22%2C%22publisher%22%3A%7B%22id%22%3A%22testpublisherid%22%7D%7D%2C%22device%22%3A%7B%22language%22%3A%22en%22%2C%22w%22%3A300%2C%22h%22%3A250%2C%22js%22%3A1%2C%22ua%22%3A%22Mozilla%2F5.0%20(Windows%20NT%2010.0%3B%20Win64%3B%20x64)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F71.0.3578.98%20Safari%2F537.36%22%7D%2C%22bidId%22%3A%224746fcaa11197f3%22%7D',
+      'data': '',
+      'options': {'withCredentials': false}
+    };
+    const validServerResponse = {
+      'body': {'id': '48640869bd9db94',
+        'seatbid': [
+          {'bid': [{'id': 'sti0a8b9aca-6080-49f6-bd63-ba54439d2359',
+            'impid': '4746fcaa11197f3',
+            'price': 3,
+            'nurl': 'https://eu.stinger-bidder.tech/api/v1/services/win?seller=prebid&price=${AUCTION_PRICE}&bidcur=USD&bidid=sti0a8b9aca-6080-49f6-bd63-ba54439d2359&strict=true',
```

Comment:
remove all these  - this should not be exposed in a public repo

Reply:
do you mean changing the nurl?
i need this test to pass code coverage.

Reply:
Why does it need to be with actual production values ?

Reply:
i will try to change the values